client/asset/btc/btc.go

```diff
@@ -522,9 +522,14 @@ func (btc *ExchangeWallet) feeRateWithFallback() uint64 {
 // FundOrder selects utxos (as asset.Coin) for use in an order. Any Coins
 // returned will be locked. Part of the asset.Wallet interface.
 func (btc *ExchangeWallet) FundOrder(value uint64, immediate bool, nfo *dex.Asset) (asset.Coins, error) {
+	btc.log.Debugf("Attempting to fund order for %d based units of %s", value, nfo.Symbol)
 	if value == 0 {
 		return nil, fmt.Errorf("cannot fund value = 0")
 	}
+
+	btc.fundingMtx.Lock()         // before getting spendable utxos from wallet
+	defer btc.fundingMtx.Unlock() // after we update the map and lock in the wallet
```

Comment:
There's a notable difference in `FundOrder` between dcr and btc. That is, btc only stores and locks in the wallet the final finding coins after optional split.  On the other hand, dcr will fund without split first and store/lock those coins, then if split is called, it does it again for the new coins from the split tx.

This last commit updates the fundingMtx locking for this change.